src/pdf/DriverOverview/DriverOverviewPdfButton.tsx

```diff
@@ -14,55 +17,69 @@ import { PdfDataFetchResponse } from "../common";
 interface DriverOverviewData {
     driverName: string;
     date: Date;
-    tableData: DriverOverviewTableData[];
+    tableData: DriverOverviewTablesData;
     message: string;
 }
 
-const compareDriverOverviewTableData = (
-    first: DriverOverviewTableData,
-    second: DriverOverviewTableData
+const compareCollectionCentres = (
+    first: DriverOverviewRowData,
+    second: DriverOverviewRowData
 ): number => {
-    if (first.name < second.name) {
+    if (first.collectionCentre < second.collectionCentre) {
         return -1;
     }
-    if (first.name > second.name) {
+    if (first.collectionCentre > second.collectionCentre) {
         return 1;
     }
     return 0;
 };
 
-type ParcelsForDelivery = (Schema["parcels"] & {
+const comparePostcodes = (first: DriverOverviewRowData, second: DriverOverviewRowData): number => {
+    if (first.address.postcode && second.address.postcode) {
+        return first.address.postcode.localeCompare(second.address.postcode);
+    }
+    if (first.address.postcode) {
+        return 1;
+    }
+    if (second.address.postcode) {
+        return -1;
+    }
+    return 0;
+};
+
+type ParcelForDelivery = Schema["parcels"] & {
     client: Schema["clients"];
+    collection_centre: Schema["collection_centres"];
     labelCount: number;
-})[];
+};
 
 type ParcelsForDeliveryResponse =
     | {
-          data: ParcelsForDelivery;
+          data: ParcelForDelivery[];
           error: null;
       }
     | {
           data: null;
           error: { type: ParcelsForDeliveryErrorType; logId: string };
       };
 
-type ParcelsForDeliveryErrorType = "parcelFetchFailed" | "noMatchingClient";
+type ParcelsForDeliveryErrorType = "parcelFetchFailed" | "noMatchingClient" | "noCollectionCentre";
 
 const getParcelsForDelivery = async (parcelIds: string[]): Promise<ParcelsForDeliveryResponse> => {
     const { data, error } = await supabase
         .from("parcels")
-        .select("*, client:clients(*), events(event_data)")
+        .select("*, client:clients(*), events(event_data), collection_centre:collection_centres(*)")
         .in("primary_key", parcelIds)
         .limit(1, { foreignTable: "clients" })
+        .limit(1, { foreignTable: "collection_centres" })
         .eq("events.new_parcel_status", "Shipping Labels Downloaded")
-        .order("timestamp", { foreignTable: "events", ascending: false });
 
```

Comment:
Do not believe we need to order by timestamp ie. most recent event as we are ordering by postcode and collection centre. I tried implementing the sort here through .order but it was very inconsistent in what it gave me so I moved to manually sorting

Reply:
Just curious how was it inconsistent?

Reply:
It basically didn't seem to be working (I wasn't able to pin down why). It seemed to be working at first but then when I selected a larger amount of data the collection centres were all in different orders as were the postcodes. My nearest guess is that something was happening later in the code to rearrange the order. If that is the case, maybe with the changes you have suggested it will work :)

Reply:
Done!